File: src/main/java/upbrella/be/store/dto/request/UpdateStoreRequest.java
package upbrella.be.store.dto.request;

import lombok.*;
import org.hibernate.validator.constraints.Range;

import javax.validation.constraints.NotBlank;
import java.util.List;

@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class UpdateStoreRequest {

    @NotBlank
    private String name;
    @NotBlank
    private String category;
    private long classificationId;
    private long subClassificationId;
    @NotBlank
    private String address;
    @NotBlank
    private String addressDetail;
    @NotBlank
    private String umbrellaLocation;
    @NotBlank
    private String businessHour;
    private String contactNumber;
    private String instagramId;
    @Range(min = -90, max = 90)
    private double latitude;
    @Range(min = -180, max = 180)
    private double longitude;
    private String content;
    private List<SingleBusinessHourRequest> businessHours;
}

File: src/main/kotlin/upbrella/be/store/dto/request/UpdateStoreRequest.kt
package upbrella.be.store.dto.request

import org.hibernate.validator.constraints.Range
import javax.validation.constraints.NotBlank

data class UpdateStoreRequest(

    @field:NotBlank
    val name: String,

    @field:NotBlank
    val category: String,

    val classificationId: Long,
    val subClassificationId: Long,

    @field:NotBlank
    val address: String,

    @field:NotBlank
    val addressDetail: String,

    @field:NotBlank
    val umbrellaLocation: String,

    @field:NotBlank
    val businessHour: String,

    val contactNumber: String?,
    val instagramId: String?,

    @field:Range(min = -90, max = 90)
    val latitude: Double,

    @field:Range(min = -180, max = 180)
    val longitude: Double,

    val content: String?,
    val businessHours: List<SingleBusinessHourRequest>
)
